connectors/cli/connector.py

```diff
@@ -0,0 +1,198 @@
+import asyncio
+from collections import OrderedDict
+
+from connectors.es.client import ESClient
+from connectors.es.settings import DEFAULT_LANGUAGE, Mappings, Settings
+from connectors.protocol import (
+    CONCRETE_CONNECTORS_INDEX,
+    CONCRETE_JOBS_INDEX,
+    ConnectorIndex,
+)
+from connectors.source import get_source_klass
+from connectors.utils import iso_utc
+
+
+class IndexAlreadyExists(Exception):
+    pass
+
+
+class Connector:
+    def __init__(self, config):
+        self.config = config
+
+        # initialize ES client
+        self.es_client = ESClient(self.config)
+
+        self.connector_index = ConnectorIndex(self.config)
+
+    async def list_connectors(self):
+        # TODO move this on top
+        try:
+            await self.es_client.ensure_exists(
+                indices=[CONCRETE_CONNECTORS_INDEX, CONCRETE_JOBS_INDEX]
+            )
+
+            connectors = []
+            async for connector in self.connector_index.all_connectors():
+                connectors.append(connector)
+
+            return connectors
+
+        # TODO catch exceptions
+        finally:
+            await self.connector_index.close()
+            await self.es_client.close()
+
+    async def ping(self):
+        if await self.es_client.ping():
+            await self.es_client.close()
```

Comment:
if you call any method after you close the connection, a new connection will be initialized. At least, this is what I see in my tests. 

I should close connections everywhere to avoid this error: 

```
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x1110d7fa0>, 663107.737258041)]']
connector: <aiohttp.connector.TCPConnector object at 0x111247430>
```
Also, I tried to make sure that one CLI command uses only one Elasticsearch session.